### PR TITLE
API 7.4 - Invoice changes

### PR DESCRIPTION
**Pull request by harshil21**
### Checklist

- [x] Added ``.. versionadded:: NEXT.VERSION``, ``.. versionchanged:: NEXT.VERSION`` or ``.. deprecated:: NEXT.VERSION`` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [ ] Documented code changes according to the `CSI standard <https://standards.mousepawmedia.com/en/stable/csi.html>`__
- [ ] Added myself alphabetically to ``AUTHORS.rst`` (optional)
- [ ] Added new classes & modules to the docs and all suitable ``__all__`` s
- [ ] Checked the `Stability Policy <https://docs.python-telegram-bot.org/stability_policy.html>`_ in case of deprecations or changes to documented behavior

**If the PR contains API changes (otherwise, you can ignore this passage)**

- [x] Checked the Bot API specific sections of the [Stability Policy](https://docs.python-telegram-bot.org/stability_policy.html)

- New classes:

   - [ ] Added ``self._id_attrs`` and corresponding documentation
   - [ ] ``__init__`` accepts ``api_kwargs`` as kw-only

- Added new shortcuts:

   - [ ]  In :class:`~telegram.Chat` & :class:`~telegram.User` for all methods that accept ``chat/user_id``
   - [ ]  In :class:`~telegram.Message` for all methods that accept ``chat_id`` and ``message_id``
   - [ ]  For new :class:`~telegram.Message` shortcuts: Added ``do_quote`` argument if methods accepts ``reply_to_message_id``
   - [ ] In :class:`~telegram.CallbackQuery` for all methods that accept either ``chat_id`` and ``message_id`` or ``inline_message_id``

-  If relevant:

   - [ ] Added new constants at :mod:`telegram.constants` and shortcuts to them as class variables
   - [ ] Link new and existing constants in docstrings instead of hard-coded numbers and strings
   - [ ]  Add new message types to :attr:`telegram.Message.effective_attachment`
   - [ ] Added new handlers for new update types
   - [ ] Add the handlers to the warning loop in the :class:`~telegram.ext.ConversationHandler`
   - [ ] Added new filters for new message (sub)types
   - [x]  Added or updated documentation for the changed class(es) and/or method(s)
   - [ ] Added the new method(s) to ``_extbot.py``
   - [ ] Added or updated ``bot_methods.rst``
   - [ ] Updated the Bot API version number in all places: ``README.rst`` and ``README_RAW.rst`` (including the badge), as well as ``telegram.constants.BOT_API_VERSION_INFO``
   - [ ] Added logic for arbitrary callback data in :class:`telegram.ext.ExtBot` for new methods that either accept a ``reply_markup`` in some form or have a return type that is/contains :class:`~telegram.Message`

### CODE REVIEW

File: telegram/_bot.py [hunk 1]
```diff
@@ -4973,12 +4973,20 @@ async def send_invoice(
...
                 displayed to the user, use for your internal processes.
             provider_token (:obj:`str`): Payments provider token, obtained via
-                `@BotFather <https://t.me/BotFather>`_.
+                `@BotFather <https://t.me/BotFather>`_. Pass an empty string for payments in
+                `Telegram Stars <https://t.me/BotNews/90>`_.
+
+                .. deprecated:: NEXT.VERSION
+                    As of Bot API 7.4, this parameter is now optional and future versions of the
+                    library will make it optional as well.
+
```
Comment (Bibo-Joshi) [inline telegram/_bot.py:4982]: just wondering: TG marked this as optional (i.e. it's allowed to completely omit), but says it wants an empty string for TG stars - so it's required after all or does TG convert None/Null to ""?
Did not find the question on the bot api repo yet https://github.com/tdlib/telegram-bot-api/issues?q=is%3Aissue+is%3Aopen+provider_token

File: telegram/_bot.py [hunk 2]
```diff
@@ -5020,19 +5029,26 @@ async def send_invoice(
             photo_width (:obj:`int`, optional): Photo width.
             photo_height (:obj:`int`, optional): Photo height.
             need_name (:obj:`bool`, optional): Pass :obj:`True`, if you require the user's full
-                name to complete the order.
+                name to complete the order. Ignored for payments in `Telegram Stars
+                <https://t.me/BotNews/90>`_.
```
Comment (Bibo-Joshi) [inline telegram/_bot.py:5033]: rst substitation? maybe at least for "`Telegram Stars <https://t.me/BotNews/90>`_"?

File: telegram/_bot.py (hunk 1 above)
Comment (harshil21) [reply telegram/_bot.py:4982]: that's what I was thinking too. I passed `None` for that argument and it worked as well.

Comment (Bibo-Joshi) [reply telegram/_bot.py:4982]: https://t.me/c/1494805131/41640

> Empty string and omitting is equivalent unless otherwise documented

**Review by Bibo-Joshi — APPROVED** [review]
Thanks for the updates! merge at will :)

**Pull request merged**